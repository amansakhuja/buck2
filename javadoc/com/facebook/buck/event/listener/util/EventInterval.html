<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc -->
<title>EventInterval (Buck)</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="EventInterval (Buck)";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":9,"i1":10,"i2":10,"i3":6,"i4":6,"i5":10,"i6":10,"i7":10,"i8":10,"i9":9,"i10":9,"i11":9,"i12":10,"i13":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],4:["t3","Abstract Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">com.facebook.buck.event.listener.util</a></div>
<h2 title="Class EventInterval" class="title">Class EventInterval</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>com.facebook.buck.event.listener.util.EventInterval</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public abstract class <span class="typeNameLabel">EventInterval</span>
extends <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a></pre>
<div class="block">Utility class to help match up start and end events</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">EventInterval</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t3" class="tableTab"><span><a href="javascript:show(4);">Abstract Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static <a href="EventInterval.html" title="class in com.facebook.buck.event.listener.util">EventInterval</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#finish(long)">finish</a></span>&#8203;(long&nbsp;finish)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>long</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getElapsedTimeMs()">getElapsedTimeMs</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>long</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getEndTime()">getEndTime</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>abstract <a href="http://docs.oracle.com/javase/7/docs/api/java/util/OptionalLong.html?is-external=true" title="class or interface in java.util" class="externalLink">OptionalLong</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFinish()">getFinish</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>abstract <a href="http://docs.oracle.com/javase/7/docs/api/java/util/OptionalLong.html?is-external=true" title="class or interface in java.util" class="externalLink">OptionalLong</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getStart()">getStart</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>long</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getStartTime()">getStartTime</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isComplete()">isComplete</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isOngoing()">isOngoing</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isStarted()">isStarted</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>static <a href="EventInterval.html" title="class in com.facebook.buck.event.listener.util">EventInterval</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#of(java.util.OptionalLong,java.util.OptionalLong)">of</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/OptionalLong.html?is-external=true" title="class or interface in java.util" class="externalLink">OptionalLong</a>&nbsp;start,
  <a href="http://docs.oracle.com/javase/7/docs/api/java/util/OptionalLong.html?is-external=true" title="class or interface in java.util" class="externalLink">OptionalLong</a>&nbsp;end)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>static <a href="EventInterval.html" title="class in com.facebook.buck.event.listener.util">EventInterval</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#proxy(long,long)">proxy</a></span>&#8203;(long&nbsp;start,
     long&nbsp;end)</code></th>
<td class="colLast">
<div class="block">Build a proxy event pair from a start and end timestamp</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>static <a href="EventInterval.html" title="class in com.facebook.buck.event.listener.util">EventInterval</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#start(long)">start</a></span>&#8203;(long&nbsp;start)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code><a href="EventInterval.html" title="class in com.facebook.buck.event.listener.util">EventInterval</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#withFinish(long)">withFinish</a></span>&#8203;(long&nbsp;timestampMillis)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code><a href="EventInterval.html" title="class in com.facebook.buck.event.listener.util">EventInterval</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#withStart(long)">withStart</a></span>&#8203;(long&nbsp;timestampMillis)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang" class="externalLink">clone</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang" class="externalLink">equals</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang" class="externalLink">finalize</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang" class="externalLink">getClass</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang" class="externalLink">hashCode</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang" class="externalLink">notify</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang" class="externalLink">notifyAll</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang" class="externalLink">toString</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang" class="externalLink">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang" class="externalLink">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long,int)" title="class or interface in java.lang" class="externalLink">wait</a></code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>EventInterval</h4>
<pre>public&nbsp;EventInterval()</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="getStart()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getStart</h4>
<pre class="methodSignature">public abstract&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/OptionalLong.html?is-external=true" title="class or interface in java.util" class="externalLink">OptionalLong</a>&nbsp;getStart()</pre>
</li>
</ul>
<a id="getFinish()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFinish</h4>
<pre class="methodSignature">public abstract&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/OptionalLong.html?is-external=true" title="class or interface in java.util" class="externalLink">OptionalLong</a>&nbsp;getFinish()</pre>
</li>
</ul>
<a id="isStarted()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isStarted</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;isStarted()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>true if this event pair has a start.</dd>
</dl>
</li>
</ul>
<a id="isComplete()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isComplete</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;isComplete()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>true if this event pair has a start and an end, false otherwise.</dd>
</dl>
</li>
</ul>
<a id="isOngoing()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isOngoing</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;isOngoing()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>true if this event pair has been started, but has not yet been finished.</dd>
</dl>
</li>
</ul>
<a id="getStartTime()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getStartTime</h4>
<pre class="methodSignature">public&nbsp;long&nbsp;getStartTime()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the start time of this event or -1 if this pair does not contain a start</dd>
</dl>
</li>
</ul>
<a id="getEndTime()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getEndTime</h4>
<pre class="methodSignature">public&nbsp;long&nbsp;getEndTime()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the end time of this event or -1 if this pair does not contain an end</dd>
</dl>
</li>
</ul>
<a id="getElapsedTimeMs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getElapsedTimeMs</h4>
<pre class="methodSignature">public&nbsp;long&nbsp;getElapsedTimeMs()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the difference between the start and end events in ms if this event pair is complete, 0
     otherwise.</dd>
</dl>
</li>
</ul>
<a id="proxy(long,long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>proxy</h4>
<pre class="methodSignature">public static&nbsp;<a href="EventInterval.html" title="class in com.facebook.buck.event.listener.util">EventInterval</a>&nbsp;proxy&#8203;(long&nbsp;start,
                                  long&nbsp;end)</pre>
<div class="block">Build a proxy event pair from a start and end timestamp</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>start</code> - the start time of the resulting pair</dd>
<dd><code>end</code> - the end time of the resulting pair</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>an event pair made from two proxy (synthetic) events</dd>
</dl>
</li>
</ul>
<a id="of(java.util.OptionalLong,java.util.OptionalLong)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>of</h4>
<pre class="methodSignature">public static&nbsp;<a href="EventInterval.html" title="class in com.facebook.buck.event.listener.util">EventInterval</a>&nbsp;of&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/OptionalLong.html?is-external=true" title="class or interface in java.util" class="externalLink">OptionalLong</a>&nbsp;start,
                               <a href="http://docs.oracle.com/javase/7/docs/api/java/util/OptionalLong.html?is-external=true" title="class or interface in java.util" class="externalLink">OptionalLong</a>&nbsp;end)</pre>
</li>
</ul>
<a id="start(long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>start</h4>
<pre class="methodSignature">public static&nbsp;<a href="EventInterval.html" title="class in com.facebook.buck.event.listener.util">EventInterval</a>&nbsp;start&#8203;(long&nbsp;start)</pre>
</li>
</ul>
<a id="finish(long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>finish</h4>
<pre class="methodSignature">public static&nbsp;<a href="EventInterval.html" title="class in com.facebook.buck.event.listener.util">EventInterval</a>&nbsp;finish&#8203;(long&nbsp;finish)</pre>
</li>
</ul>
<a id="withStart(long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>withStart</h4>
<pre class="methodSignature">public&nbsp;<a href="EventInterval.html" title="class in com.facebook.buck.event.listener.util">EventInterval</a>&nbsp;withStart&#8203;(long&nbsp;timestampMillis)</pre>
</li>
</ul>
<a id="withFinish(long)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>withFinish</h4>
<pre class="methodSignature">public&nbsp;<a href="EventInterval.html" title="class in com.facebook.buck.event.listener.util">EventInterval</a>&nbsp;withFinish&#8203;(long&nbsp;timestampMillis)</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
