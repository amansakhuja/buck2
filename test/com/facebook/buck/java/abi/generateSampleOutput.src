package com.facebook.buck.example;

import java.util.Set;

public final class A<T> {

  // A field that defines a constant, but which is private and should be missing.
  private final static String HIDDEN = "yo";
   // But we do expect this constant to contain the value of "HIDDEN".
  public final static String VISIBLE = HIDDEN + " another literal ";

   // We expect LOCK, but no constant value declaration.
  public static final Object LOCK = new Object();

  // A constant. Because the value of LOCK.toString() is calculated at runtime, the value is
  // omitted.
  public final static String VISIBLE_BUT_WEIRD = HIDDEN + LOCK.toString();

  public void doSomething(String... args) {
  }

  public void foo(Object... foos) {
  }

  public void foo(Object[] foo, Object... foos) {
  }

  // Weird classes defined in methods should not leak out into the ABI.
  public Runnable getThing() {
    class Example implements Runnable {
      public long LONG_NUMBER = Long.MAX_VALUE;
      @Override
      public void run() { }
    }
    return new Example();
  }

  // Ensure that we can nest generics.
  Set<Set<String>> someStrings;

  // And that we correctly report generics when solely defined on the method.
  <X> X getThing(T value) { return null; }

  // A constant, out of order from the others, that should have the value included.
  protected final static int MAGIC = 42;

  // We should keep the deprecated annotation.
  @Deprecated
  protected T doSomethingElse() {
    System.out.println('c'); return null;
  }

  // There's some debate about whether to include overridden methods from parent classes.
  // Include them for for now.
  @Override public String toString() { return super.toString() + "42"; }

  // A nested inner class
  private static class Inner {
    public static final int MEANING_OF_LIFE = 42;
    public String getFoo() { return "foo"; }
  }

  public Inner getInner() { return new Inner(); }

  // Demonstrate nesting to an arbitrary depth
  public static class Outer {
    public class InstanceOuter {
      Outer getOuter() { return Outer.this; }
    }
    Outer getOuter() { return new Outer(); }
  }
}
