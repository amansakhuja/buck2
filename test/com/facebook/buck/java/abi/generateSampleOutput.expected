public final class com.facebook.buck.example.A<T> extends java.lang.Object
<X> X getThing(T)
@java.lang.Deprecated protected T doSomethingElse()
@java.lang.Override public java.lang.String toString()
java.util.Set<java.util.Set<java.lang.String>> someStrings
protected static final int MAGIC = 42
public A()
public com.facebook.buck.example.A.Inner getInner()
public java.lang.Runnable getThing()
public static final java.lang.Object LOCK
public static final java.lang.String VISIBLE = yo another literal 
public static final java.lang.String VISIBLE_BUT_WEIRD
public void doSomething(java.lang.String...)
public void foo(java.lang.Object...)
public void foo(java.lang.Object[], java.lang.Object...)
-----
public static class com.facebook.buck.example.A.Outer extends java.lang.Object
com.facebook.buck.example.A.Outer getOuter()
public Outer()
-----
public class com.facebook.buck.example.A.Outer.InstanceOuter extends java.lang.Object
com.facebook.buck.example.A.Outer getOuter()
public InstanceOuter()
